.meta/mkcompletion.py

```diff
@@ -6,13 +6,48 @@
 import tqdm.cli
 from io import open as io_open
 from os import path
+import re
+
+def doc2opt(doc):
+    """
+    doc  : str, document to parse
+    """
+    options = ['--' + i.strip('  :') for i in re.findall(r'\w+  :', doc)]
+    return options
+
+# CLI options
+options = ['-h', '--help', '-v', '--version']
+options += doc2opt(tqdm.tqdm.__init__.__doc__)
+options += doc2opt(tqdm.cli.CLI_EXTRA_DOC)
+
+# Remove options without CLI support
+no_support = ['--' + i for i in tqdm.cli.UNSUPPORTED_OPTS]
+# TODO: Check and remove option `name`
+no_support += ['--name']
+options = list(set(options) - set(no_support))
 
 src_dir = path.abspath(path.dirname(__file__))
 completion = u"""\
 #!/usr/bin/env bash
 
-# TODO: programatically generate based on tqdm.tqdm/tqdm.cli
-"""
+_tqdm()
+{{
+    local cur prv
+    cur="${{COMP_WORDS[COMP_CWORD]}}"
+    prv="${{COMP_WORDS[COMP_CWORD - 1]}}"
+
+    case ${{prv}} in
+        "--log")
```

Comment:
Should the arguments have auto complete as well. That would require some additional logic to extract the acceptable options from the docstring. Manually added for `--log` parameter.

Reply:
yes, ideally things with a defined set of options should auto-complete properly. If you think that's too hard could leave it for a separate issue later.

Reply:
As a first, we could switch off completion when `$prv` matches an option from `\w+  : (str|int|float|chr|dict|tuple)`.

Reply:
We would have to match or keep track of the argument to which the options belong to. Maybe have to change the `findall()` to a line by line loop based search. Any ideas?

Reply:
Yes will do

Reply:
done